readthedocs/projects/tasks.py

```diff
@@ -335,11 +337,75 @@ def setup_vcs(self):
         self.setup_env.update_build(state=BUILD_STATE_CLONING)
 
         self._log(msg='Updating docs from VCS')
-        update_imported_docs(self.version.pk)
+        self.sync_repo()
         commit = self.project.vcs_repo(self.version.slug).commit
         if commit:
             self.build['commit'] = commit
 
+    def sync_repo(self):
+        """
+        Checkout/update the project's repository and hit ``sync_versions`` API.
+        """
+        # Make Dirs
+        if not os.path.exists(self.project.doc_path):
+            os.makedirs(self.project.doc_path)
+
+        if not self.project.vcs_repo():
+            raise RepositoryError(
+                _('Repository type "{repo_type}" unknown').format(
+                    repo_type=self.project.repo_type,
+                ),
+            )
+
+        with self.project.repo_nonblockinglock(
+                version=self.version,
+                max_lock_age=getattr(settings, 'REPO_LOCK_SECONDS', 30)):
+
+            # Get the actual code on disk
+            try:
+                before_vcs.send(sender=self.version)
+                self._log(
+                    'Checking out version {slug}: {identifier}'.format(
+                        slug=self.version.slug,
+                        identifier=self.version.identifier,
+                    ),
+                )
+                version_repo = self.project.vcs_repo(
+                    self.version.slug,
+                    # When ``sync_only`` we don't a setup_env
+                    getattr(self, 'setup_env', None),
```

Comment:
This is also kind of hacky: `sync_repo` can be executed before (sync_only=True) or after the `run_setup` (the one that creates the `self.setup_env` we want to use here in case it exists)